plugins/woocommerce/includes/emails/class-wc-email-customer-invoice.php

```diff
@@ -52,11 +54,7 @@ public function __construct() {
 		 * @return string
 		 */
 		public function get_default_subject( $paid = false ) {
-			if ( $paid ) {
-				return __( 'Invoice for order #{order_number} on {site_title}', 'woocommerce' );
-			} else {
-				return __( 'Your latest {site_title} invoice', 'woocommerce' );
-			}
+			return __( 'Order #{order_number} from {site_title}', 'woocommerce' );
```

Comment:
```suggestion
			return __( 'Details for order #{order_number} on {site_title}', 'woocommerce' );
```

Via [feedback](https://github.com/woocommerce/woocommerce/pull/45395#issuecomment-1983913935) from @webdados.